src/Raven.Server/Documents/Queries/Dynamic/CollectionQueryRunner.cs

```diff
@@ -295,12 +295,21 @@ private unsafe void FillCountOfResultsAndIndexEtag(QueryResultServerSide<Documen
             var collection = query.CollectionName;
             var buffer = stackalloc long[bufferSize];
 
+            long numberOfDocuments;
+            
+            if (collection == Constants.Documents.Collections.AllDocumentsCollection)
```

Comment:
would prefer to compare this string once, introduce a bool isAllDocumentsCollection and use it here and few lines down

Reply:
Changed